src/core/shape/vertex.js

```diff
@@ -2253,4 +2253,177 @@ p5.prototype.normal = function(x, y, z) {
+ *  }
+ * `;
+ * 
+ * function setup(){
+ *   createCanvas(100, 100, WEBGL);
+ *
+ *   // Create and use the custom shader.
+ *   const myShader = createShader(vertSrc, fragSrc);
+ *   shader(myShader);
+ * 
+ *   describe('A wobbly, cyan circle on a gray background.');
+ * }
+ * 
+ * function draw(){
+ *   // Set the styles
+ *   background(125);
+ *   noStroke();
+ * 
+ *   // Draw the circle.
+ *   beginShape();
+ *   for (let i = 0; i < 30; i++){
+ *     const x = 40 * cos(i/30 * TWO_PI);
+ *     const y = 40 * sin(i/30 * TWO_PI);
+ *
+ *     // Apply some noise to the coordinates.
+ *     const xOff = 10 * noise(x + millis()/1000) - 5;
+ *     const yOff = 10 * noise(y + millis()/1000) - 5;
+ * 
+ *     // Apply these noise values to the following vertex.
+ *     setAttribute('aOffset', [xOff, yOff]);
+ *     vertex(x, y);
+ *   }
+ *   endShape(CLOSE);
+ * }
+ * </code>
+ * </div>
+ * 
+ * <div>
+ * <code>
+ * let myShader;
+ * const cols = 10;
+ * const rows = 10;
+ * const cellSize = 6;
+ * 
+ * const vertSrc = `#version 300 es
+ *   precision mediump float;
+ *   uniform mat4 uProjectionMatrix;
+ *   uniform mat4 uModelViewMatrix;
+ * 
+ *   in vec3 aPosition;
+ *   in vec3 aNormal;
+ *   in vec3 aVertexColor;
+ *   in float aDistance;
+ * 
+ *   out vec3 vVertexColor;
+ *   
+ *   void main(){
+ *     vec4 positionVec4 = vec4(aPosition, 1.0);
+ *     positionVec4.xyz += aDistance * aNormal * 2.0;;
+ *     vVertexColor = aVertexColor;
+ *     gl_Position = uProjectionMatrix * uModelViewMatrix * positionVec4;
+ *   }
+ * `;
+ * 
+ * const fragSrc = `#version 300 es
+ *   precision mediump float;
+ *   
+ *   in vec3 vVertexColor;
+ *   out vec4 outColor;
+ *   
+ *   void main(){
+ *     outColor = vec4(vVertexColor, 1.0);
+ *   }
+ * `;
+ * 
+ * function setup(){
+ *   createCanvas(100, 100, WEBGL);
+ * 
+ *   // Create and apply the custom shader.
+ *   myShader = createShader(vertSrc, fragSrc);
+ *   shader(myShader);
+ *   noStroke();
+ *   describe('A blue grid, which moves away from the mouse position, on a gray background.');
+ * }
+ * 
+ * function draw(){
+ *   background(200);
+ * 
+ *   // Draw the grid in the middle of the screen.
+ *   translate(-cols*cellSize/2, -rows*cellSize/2);
+ *   beginShape(QUADS);
+ *   for (let i = 0; i < cols; i++) {
+ *     for (let j = 0; j < rows; j++) {
+ * 
+ *       // Calculate the cell position.
+ *       let x = i * cellSize;
+ *       let y = j * cellSize;
+ * 
+ *       fill(j/rows*255, j/cols*255, 255);
+ * 
+ *       // Calculate the distance from the corner of each cell to the mouse.
+ *       let distance = dist(x1,y1, mouseX, mouseY);
+ * 
+ *       // Send the distance to the shader.
+ *       setAttribute('aDistance', min(distance, 100));
+ * 
+ *       vertex(x, y);
+ *       vertex(x + cellSize, y);
+ *       vertex(x + cellSize, y + cellSize);
+ *       vertex(x, y + cellSize);
+ *     }
+ *   }
+ *   endShape();
+ * }
+ * </code>
+ * </div>
+/
+/**
```

Comment:
I think we might need these to all be in one doc comment so that they don't show up as two separate overloads. Does it work if we remove this line and the one above?